File: src/io/github/mg138/tsbook/items/data/Stat.java
package io.github.mg138.tsbook.items.data;

import java.io.Serializable;

public class Stat implements Serializable {
    public StatRange statRange;
    public double stat;

    public Stat(double stat) {
        this.stat = stat;
    }

    public Stat(StatRange statRange) {
        this.statRange = statRange;
    }
}

File: src/main/kotlin/io/github/mg138/tsbook/item/attribute/stat/Stat.kt
package io.github.mg138.tsbook.item.attribute.stat

interface Stat {
    fun getStat(): Double

    fun incompatible(that: Stat) =
        "Incompatible Stat type. (${this::class.simpleName} verses ${that::class.simpleName})"

    fun applyPlaceholder(string: String): String

    operator fun plus(increment: Stat?): Stat

    operator fun minus(decrement: Stat?): Stat

    operator fun times(multiplier: Int): Stat

    operator fun times(multiplier: Float): Stat

    operator fun times(multiplier: Double): Stat

    operator fun div(divisor: Int): Stat

    operator fun div(divisor: Float): Stat

    operator fun div(divisor: Double): Stat
}